src/actions/compiler_message_parsing.rs

```diff
@@ -55,12 +56,29 @@ pub fn parse(message: &str) -> Result<FileDiagnostic, ParseError> {
         return Err(ParseError::NoSpans);
     }
 
+    let message_text = compose_message(&message);
     let span = message.spans[0].rls_span().zero_indexed();

```

Comment:
@jonathandturner,
The first span isn't always primary one.
Is that okay for you?